server/main_server/src/rodsServer.cpp

```diff
@@ -119,66 +114,135 @@ namespace
+                rodsLog(LOG_ERROR, "createAndSetRECacheSalt: failed to set environment variable");
+                return ERROR(SYS_SETENV_ERR, "createAndSetRECacheSalt: failed to set environment variable");
             }
 
             return SUCCESS();
         }
     }
 
-    int get64RandomBytes( char *buf ) {
+    int get64RandomBytes(char* buf)
+    {
         const int num_random_bytes = 32;
-        const int num_hex_bytes = 2 * num_random_bytes;
         unsigned char random_bytes[num_random_bytes];
-        irods::getRandomBytes( random_bytes, sizeof(random_bytes) );
+        irods::getRandomBytes(random_bytes, sizeof(random_bytes));
 
         std::stringstream ss;
-        for ( size_t i = 0; i < sizeof(random_bytes); ++i ) {
-            ss << std::hex << std::setw(2) << std::setfill('0') << (unsigned int)( random_bytes[i] );
+        for (std::size_t i = 0; i < sizeof(random_bytes); ++i) {
+            ss << std::hex << std::setw(2) << std::setfill('0') << (unsigned int) random_bytes[i];
         }
 
-        snprintf( buf, num_hex_bytes + 1, "%s", ss.str().c_str() );
+        const int num_hex_bytes = 2 * num_random_bytes;
+        std::snprintf(buf, num_hex_bytes + 1, "%s", ss.str().c_str());
+
         return 0;
-    }
+    } // get64RandomBytes
+
+    bool instantiate_shared_memory_for_plugin(const nlohmann::json& _plugin_object)
+    {
+        const auto itr = _plugin_object.find(irods::CFG_SHARED_MEMORY_INSTANCE_KW);
+
+        if (_plugin_object.end() != itr) {
+            const auto& mem_name = itr->get_ref<const std::string&>();
+            prepareServerSharedMemory(mem_name);
+            detachSharedMemory(mem_name);
+            return true;
+        }
+
+        return false;
+    } // instantiate_shared_memory
+
+    irods::error instantiate_shared_memory()
+    {
+        try {
+            const auto& config = irods::server_properties::instance().map();
+
+            for (const auto& item : config.at(irods::CFG_PLUGIN_CONFIGURATION_KW).items()) {
+                for (const auto& plugin : item.value().items()) {
+                    instantiate_shared_memory_for_plugin(plugin.value());
+                }
+            }
+        }
+        catch (const boost::bad_any_cast& e) {
+            return ERROR(INVALID_ANY_CAST, e.what());
+        }
+        catch (const irods::exception& e) {
+            return irods::error(e);
+        }
+
+        return SUCCESS();
+    } // instantiate_shared_memory
+
+    bool uninstantiate_shared_memory_for_plugin(const nlohmann::json& _plugin_object)
+    {
+        const auto itr = _plugin_object.find(irods::CFG_SHARED_MEMORY_INSTANCE_KW);
+
+        if (_plugin_object.end() != itr) {
+            const auto& mem_name = itr->get_ref<const std::string&>();
+            removeSharedMemory(mem_name);
+            resetMutex(mem_name.c_str());
+            return true;
+        }
+
+        return false;
+    } // uninstantiate_shared_memory
```

Comment:
uninstantiate_shared_memory -> uninstantiate_shared_memory_for_plugin

Reply:
is this the right word?    honestly asking.    destroy?

Reply:
I think it was an attempt at making these functions feel related, which they are.

I'm not against using different words for these functions. `destroy` isn't a bad choice. If we went with that, I'd feel better about using `create` rather than `instantiate`.

Reply:
either is fine with me - @alanking gotta vote?

Reply:
No opinion from me. I'm fine with either!

Reply:
I'm going to create an issue about renaming these and leave it for a separate PR.

Reply:
Issue created: #6290